src/main/java/cope/cosmos/client/features/modules/visual/LagESPModule.java

```diff
@@ -0,0 +1,232 @@
+package cope.cosmos.client.features.modules.visual;
+
+import cope.cosmos.client.events.network.PacketEvent;
+import cope.cosmos.client.features.modules.Category;
+import cope.cosmos.client.features.modules.Module;
+import cope.cosmos.client.features.setting.Setting;
+import cope.cosmos.util.string.ColorUtil;
+import net.minecraft.network.play.server.SPacketPlayerPosLook;
+import net.minecraft.util.math.Vec3d;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+
+
+import java.util.LinkedList;
+
+import static org.lwjgl.opengl.GL11.*;
+import static org.lwjgl.opengl.GL11.glPopMatrix;
+
+
+public class LagESPModule extends Module {
+
+    public LagESPModule() {
+        super("LagESP", Category.VISUAL, "Shows a path from where you got rubber banded");
+    }
+
+    public static Setting<Float> fadeSpeed = new Setting<>("FadeSpeed", 1F, 2F, 10F, 1)
+            .setDescription("How long the rubber band should show for");
+
+    public static Setting<Float> width = new Setting<>("Width", 0.1F, 2F, 5F, 1)
+            .setDescription("The width of the lines");
+
+    public static Setting<Boolean> reverse = new Setting<>("Inverse", false);
+
+    public static Setting<fade> fadeSetting = new Setting<fade>("Fade", fade.LINEAR);
+
+
+    LinkedList<RubberBand> list = new LinkedList<>();
+
+
+
+    @SubscribeEvent
+    public void onPacket(PacketEvent event){
+        /*  when we get rubber-banded */
+        if (event.getPacket() instanceof SPacketPlayerPosLook) {
+            SPacketPlayerPosLook p = (SPacketPlayerPosLook) event.getPacket();
+
+            /* register that a rubberband happened */
+
+            list.add(new RubberBand(
+                    mc.player.getPositionVector(),
+                    new Vec3d(p.getX(), p.getY(), p.getZ()
+                    )));
+
+        }
+
+
+
+    }
+
+
+    /* stolen from breadcrumbs.. */
+    @Override
+    public void onRender3D() {
+
+
+        // Render positions
+        list.forEach(r -> {
```

Comment:
personally i would name the parameter something more specific but its not really an issue i guess